docs/Diagnostics.md

```diff
@@ -1,4 +1,75 @@
-# Diagnosing Issues using MsQuic
+# Diagnosing Issues with MsQuic
 
-TODO
+This document describes various ways to debug and diagnose issues when using MsQuic.
 
+# Logging
+
+For functional problems, generally logging is the best way to diagnose problems. MsQuic has extensive logs in the code to facilitate debugging.
+
+## Windows
+
+On Windows, MsQuic leverages [ETW](https://docs.microsoft.com/en-us/windows/win32/etw/event-tracing-portal) for its logging.
+
+To start collecting a trace, you can use the following command:
+
+```
+netsh.exe trace start overwrite=yes report=dis correlation=dis traceFile=quic.etl provider={ff15e657-4f26-570e-88ab-0796b258d11c} level=0x5 keywords=0xffffffff
+```
+
+And to stop log the trace session, you can use the following command:
+
+```
+netsh.exe trace stop
+```
+
+To convert the trace, you can use the following command:
+
+```
+netsh.exe trace convert quic.etl
+```
+
+> **Important** - If you're using a version of MsQuic that uses an ETW manifest version more recent than the one built into the Windows image, decoding may not provide correct output. **TODO** - Provide instructions to get around this problem.
+
+## Linux
+
+On Linux, MsQuic leverages [LTTng](https://lttng.org/features/) for its logging.
+
+To start collecting a trace, you can use the following commands:
+
+```
+mkdir msquic_lttng
+lttng create msquic -o=./msquic_lttng
+lttng enable-event --userspace CLOG_*
+lttng add-context --userspace --type=vpid --type=vtid
+lttng start
+```
+
+And to stop log the trace session, you can use the following command:
+
+```
+lttng stop msquic
+```
+
+To convert the trace, you can use the following commands:
+
+```
+babeltrace --names all ./msquic_lttng/* > quic.babel.txt
+clog2text_lttng -i quic.babel.txt -s clog.sidecar -o quic.log --showTimestamp --showCpuInfo
```

Comment:
Probably need it documented that you need clog2text_lttng installed. Just pointing to prepare-machine.ps1 is probably fine.

Reply:
Yeah, there's a lot more that needs to be added. For now, I don't want to add any more. If you're feeling up to it, feel free to add more commits to this PR/branch.